hazelcast/src/test/java/com/hazelcast/sql/impl/expression/datetime/FormatterTest.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.datetime;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.ZoneOffset;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+@SuppressWarnings("checkstyle:ParenPad")
+public class FormatterTest {
+    private static int jdk;
```

Comment:
More failing tests:
```java
    @Test
    public void testEmpty() {
        Formatter f = new Formatter("");
        check(1, f, "");
        f = new Formatter("th");
        check(1, f, "st");
        f = new Formatter("SG");
        check(1, f, "+");
        check(-1, f, "-");
    }

    @Test
    public void testNoIntegerPart() {
        Formatter f = new Formatter(".00");
        check(1, f, " .##");
        check(0.5, f, " .50");
    }
```

Reply:
`th` alone causes an overflow since there is no integer digit. The rest is covered, except for `.00` (trailing zeros when there is no integer part). However, leading/trailing zeros are covered in `testDigitsAndGrouping()`.